sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.cpp

```diff
@@ -694,12 +698,54 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueMemBufferCopyRect(
   return Result;
 }
 
+// CUDA has no memset functions that allow setting values more than 4 bytes. UR
+// API lets you pass an arbitrary "pattern" to the buffer fill, which can be
+// more than 4 bytes. We must break up the pattern into 1 byte values, and set
+// the buffer using multiple strided calls.  The first 4 patterns are set using
+// cuMemsetD32Async then all subsequent 1 byte patterns are set using
+// cuMemset2DAsync which is called for each pattern.
+ur_result_t commonMemSetLargePattern(CUstream Stream, uint32_t PatternSize,
```

Comment:
this is ported over from the [hip adapter](https://github.com/intel/llvm/pull/9617/files#diff-6636344b5f9ab5be80c4fa9939d0715b8ddcbdfbcd493d0d150438d2378f28f5R700)